src/main/resources/data/computercraft/lua/rom/programs/poster.lua

```diff
@@ -27,68 +37,137 @@ if args[1] == "stop" then
+  local stream
 
-local function load2djFile(name)
-  local f
   if http and name:match("^https?://") then
     print("Downloading...")
-    f = http.get{ url = name, binary = true }
+    stream = http.get{ url = name, binary = true }
   else
+    print("Loading...")
     name = shell.resolve(name)
     if not fs.exists(name) then
-      -- Try to find a .2dj file if the user didn't specify the extension
-      if not filename:match(".2dj$") then
-        local newName = name .. ".2dj"
-        if fs.exists(newName) then
-          return load2djFile(newName)
-        end
-      end
-      error("File not found: " .. name, 0)
+      error(("File \'%s\' not found"):format(name), 0)
     end
-    f = fs.open(name, "r")
+    stream = fs.open(name, "r")
   end
   
-  if not f then
-    error("Could not open " .. name, 0)
+  if not stream then
+    error(("Could not open \'%s\'"):format(name), 0)
   end
-  local data = f.readAll()
-  f.close()
+
+  local data = stream.readAll()
+  stream.close()
 
   local data, err = textutils.unserialiseJSON(data)
-  if data == nil then
-    error("\n\nCould not parse " .. name .. ": " .. err, 0)
+  if type(data) ~= "table" then
+    error(("\n\nCould not parse \'%s\': %s"):format(name, err), 0)
   end
   return data
 end
 
-local printData = load2djFile(filename)
-if not printData.pixels then
-  error("Invalid print data - missing pixels")
+-- Check if given poster is valid
+local function isValidPoster(data)
+  if type(data.pixels) ~= "table" then
+    return true, "Table \'pixels\' does not exist"
+  elseif type(data.palette) ~= "table" then
+    return true, "Table \'palette\' does not exist"
+  end
+
+  return false
 end
 
--- Output some basic print metadata (models often have credit in tooltips)
-term.write("Printing: ")
-term.setTextColor(colors.lightGray)
-print(printData.label or filename)
+local posters = {}
+local data = loadPoster(filename)
+-- Add all needed pages to stack (.2dja)
+if type(data.pages) == "table" then
+  if index.all then
+    index.from = 1
+    index.to = #data.pages
+  elseif index.to > #data.pages then
+    error(("Index out of bounds (got %d, limit: %d)"):format(index.to, #data.pages), 0)
+  end
 
-if printData.tooltip then
-  term.write(printData.tooltip)
+  for i=index.from, index.to do
+    local poster = data.pages[i]
+    local hasError, msg = isValidPoster(poster)
+    if hasError then
+      error(("Could not load page %d: \'%s\'"):format(i, msg), 0)
+    end
+
+    table.insert(posters, poster)
+  end
+else
+  -- Add poster to stack x nCounter (.2dj)
+  local hasError, msg = isValidPoster(data)
+  if hasError then
+    error(("Could not load .2dj file: \'%s\'"):format(msg), 0)
+  end
+  table.insert(posters, data)
 end
 
-print("")
+-- Setup colors
+local highlightColor = colors.lightGray
+local errColor = colors.gray
+local greenColor = colors.white
+
+if term.isColor() then
+  highlightColor = colors.yellow
+  errColor = colors.red
+  greenColor = colors.green
+end
```

Comment:
I think this is the wrong place to do this, I would prefer the colors be calculated at the top of the file (right next to `local startTColor = term.getTextColor()`)